api/hooksystem.ts

```diff
@@ -0,0 +1,52 @@
+import type { APActivity } from 'activitypub-types'
+import type { BasicFetchParams, FetchLike } from './apsystem'
+import { HookStore } from '../store/HookStore'
+
+export default class HookSystem {
+  store: HookStore
+  fetch: FetchLike
+
+  constructor (store: HookStore, fetch: FetchLike = globalThis.fetch) {
```

Comment:
```suggestion
  constructor (store: Store, fetch: FetchLike = globalThis.fetch) {
```

Hook stores are actor specific so we should take the full `store` and use `store.forActor(actor).hooks`